src/CHIPDriver.hh

```diff
@@ -118,4 +120,171 @@ extern std::atomic_ulong CHIPNumRegisteredFatBinaries;
  */
 extern thread_local hipError_t CHIPTlsLastError;
 
+class DeviceType {
+public:
+  enum Type { GPU, CPU, ACCEL, FPGA, DEFAULT };
+
+private:
+  Type Type_;
+
+public:
+  DeviceType(Type TypeIn) : Type_(TypeIn) {}
+
+  std::string str() const {
+    switch (Type_) {
+    case GPU:
+      return "gpu";
+    case CPU:
+      return "cpu";
+    case ACCEL:
+      return "accel";
+    case FPGA:
+      return "fpga";
+    case DEFAULT:
+      return "default";
+    default:
+      return "unknown";
+    }
+  }
+
+  Type getType() const { return Type_; }
+};
```

Comment:
Seems excessive. This could be simplified to enum class and free-standing `const std::string& toString(DeviceType Type)`.

Reply:
I don't see any benefit to having related functions outside of the class definition.